test/components/preprocessors/test_recursive_chunker.py

```diff
@@ -0,0 +1,198 @@
+    assert (
+        chunks[3]
+        == "Some high-profile applications include advanced web search engines (e.g., Google Search); recommendation systems (used by YouTube, Amazon, and Netflix); interacting via human speech (e.g., Google Assistant, Siri, and Alexa); autonomous vehicles (e.g., Waymo); generative and creative tools (e.g., ChatGPT and AI art); and superhuman play and analysis in strategy games (e.g., chess and Go)."
+    )  # noqa: E501
+
+
+def test_recursive_splitter_empty_documents():
+    chunker = RecursiveChunker(chunk_size=20, chunk_overlap=0, separators=["."])
+    empty_doc = Document(content="")
+    doc_chunks = chunker.run([empty_doc])
+    doc_chunks = doc_chunks["documents"]
+    assert len(doc_chunks) == 0
+
+
+def test_recursive_chunker_with_multiple_separators_recursive():
+    chunker = RecursiveChunker(
+        chunk_size=260, chunk_overlap=0, separators=["\n\n", "\n", ".", " "], keep_separator=True
+    )
+    text = """Artificial intelligence (AI) - Introduction
+
+AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.
+AI technology is widely used throughout industry, government, and science. Some high-profile applications include advanced web search engines; recommendation systems; interacting via human speech; autonomous vehicles; generative and creative tools; and superhuman play and analysis in strategy games."""  # noqa: E501
+
+    doc = Document(content=text)
+    doc_chunks = chunker.run([doc])
+    doc_chunks = doc_chunks["documents"]
+    assert len(doc_chunks) == 4
+    assert (
+        doc_chunks[0].meta["original_id"]
+        == doc_chunks[1].meta["original_id"]
+        == doc_chunks[2].meta["original_id"]
+        == doc_chunks[3].meta["original_id"]
+        == doc.id
+    )
+    assert doc_chunks[0].content == "Artificial intelligence (AI) - Introduction\n\n"
+    assert (
+        doc_chunks[1].content
+        == "AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.\n"
+    )
+    assert doc_chunks[2].content == "AI technology is widely used throughout industry, government, and science."
+    assert (
+        doc_chunks[3].content
+        == " Some high-profile applications include advanced web search engines; recommendation systems; interacting via human speech; autonomous vehicles; generative and creative tools; and superhuman play and analysis in strategy games."
+    )
+
+
+@pytest.mark.parametrize("chunk_overlap", [0, 9])
+def test_recursive_chunker_split_document_with_overlap(chunk_overlap):
+    chunker = RecursiveChunker(chunk_size=20, chunk_overlap=chunk_overlap, separators=[".", " "], keep_separator=True)
+    text = """A simple sentence.A simple sentence.A simple sentence.A simple sentence"""
+
+    doc = Document(content=text)
+    doc_chunks = chunker.run([doc])
+    doc_chunks = doc_chunks["documents"]
+    if chunker.chunk_overlap == 20:
```

Comment:
We're checking for 20 here but parametrize only sets it to 0 and 9.
Let's not use `if chunker.chunk_overlap == ...` but instead set the expected value in parametrize too, for example:
`@pytest.mark.parametrize("chunk_overlap", "expected_content", [(0, "A simple sentence."), (9, "..."])` or in this particular case: create two separate tests.

Reply:
good catch, this was a bug/leftover from testing something